File: src/my_ude_cpp/src/check_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "ude_interface/srv/odd_even_check.hpp"

using std::placeholders::_1;
using std::placeholders::_2;

class CheckServerNode : public rclcpp::Node 
{
public:
    CheckServerNode() : Node("check_server") 
    {
        service_server_ = this->create_service<ude_interface::srv::OddEvenCheck>("odd_even_check", 
                std::bind(&CheckServerNode::check_num_odd_even, this, _1, _2));
    	RCLCPP_INFO(this->get_logger(), "Check Server has been started.");
    }

private:
    void check_num_odd_even(const ude_interface::srv::OddEvenCheck::Request::SharedPtr request,
                            ude_interface::srv::OddEvenCheck::Response::SharedPtr response)
        {
            int remainder = std::abs(request->number % 2);
            switch(remainder) {
                case 0 :
                    response->decision = "Even";
                    break;
                case 1 :
                    response->decision = "Odd";
            }
            RCLCPP_INFO(this->get_logger(), "the checked number is %d and the answer is %s", int(request->number), response->decision.c_str());
        } 
    
    rclcpp::Service<ude_interface::srv::OddEvenCheck>::SharedPtr service_server_;

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<CheckServerNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_ude_py/scripts/check_server.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from ude_interface.srv import OddEvenCheck


class CheckServerNode(Node): 
    def __init__(self):
        super().__init__("check_server") 
        self.srv = self.create_service(OddEvenCheck, 'odd_even_check', self.determine_odd_even)
        self.get_logger().info("Check server has been started")

    def determine_odd_even(self, request, response): 
        if request.number % 2 == 0:
            response.decision = "Even"
        elif request.number % 2 == 1:
            response.decision = "Odd"
        else:
            response.decision = "Error"
        self.get_logger().info("the checked number is " + str(request.number) + " and the answer is " + response.decision)
        return response 


def main(args=None):
    rclpy.init(args=args)
    node = CheckServerNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()